### ISSUE TITLE
Migrating from UIWebView to WKWebView

### ISSUE THREAD

**Issue by OmarPedraza** [muxinc/mux-stats-sdk-avplayer#32]
From the [docs](https://developer.apple.com/documentation/uikit/uiwebview):

> In apps that run in iOS 8 and later, use the WKWebView class instead of using UIWebView.

Are there any plans to migrate to using WKWebView instead of UIWebView within the framework?

Your dependency GoogleAds-IMA-iOS-SDK also uses UIWebView instead of WKWebView but, as @jsanford8 said [here](https://github.com/muxinc/mux-stats-sdk-avplayer/issues/15#issuecomment-436892015), it shouldn't be a problem since that dependency would be removed.

Thanks!

**Comment by markhomoki** [comment]
Any update on this?

**Comment by jsanford8** [comment]
Hey @markhomoki - we're actually working on this right now, to remove the unused UIWebView (and other components). We should have an update soon.

**Comment by jsanford8** [comment]
In fact, this should be fixed in this release: https://github.com/muxinc/mux-stats-sdk-avplayer/releases/tag/v1.3.6

**Comment by markhomoki** [comment]
Thanks for the quick reply. Checking it now.

**Comment by markhomoki** [comment]
Unfortunately, after the update I still got an email from Apple regarding the deprecation.
Running `grep -r "UIWebView" .` in the ios folder still prints `Binary file ./Pods/Mux-Stats-AVPlayer/Frameworks/iOS/fat/MUXSDKStats.framework.dSYM/Contents/Resources/DWARF/MUXSDKStats matches`. Any idea? Thank you.

**Comment by dylanjha** [comment]
@markhomoki hmmm. You're running this in your application, correct?

Before 1.3.6 in my DemoApp I saw this:

```
❯ grep -r "UIWebView" .
Binary file ./Pods/Mux-Stats-AVPlayer/Frameworks/iOS/fat/MUXSDKStats.framework.dSYM/Contents/Resources/DWARF/MUXSDKStats matches
```

And after upgrading to 1.3.6 I didn't get any matches.

Let me see whats going on, can you run `pod deintegrate && pod install` and grep again, just to make sure the latest 1.3.6 version is getting pulled in?

**Comment by markhomoki** [comment]
Yes, that's correct.
Even after deintegrate and install, it prints the same line.
In my `Podfile.lock` file it shows the right version, so I am not entirely sure.
```
- Mux-Stats-AVPlayer (1.3.6):
  - Mux-Stats-Core (~> 2.2.0)
- Mux-Stats-Core (2.2.2)
```

**Comment by dylanjha** [comment]
Ah, I see why I was getting different results, when I installed the Pod in my DemoApp as a development pod with `:path =>` then it doesn't show up in the project, so it doesn't return any results when grepping.

I do see now the line in the `/fat/` framework in `Pods/` that is matching.

I would think that when you build your app for release it would pull in the `/release/` framework instead of the `/fat` framework.

The `dSYM` files are only debug symbols, I see that it's showing up for some other [SDKs like Mapbox in the dSYM files](https://github.com/mapbox/ios-sdk-examples/issues/373), it doesn't have code that actually gets run so I wonder if Apple is being a little aggressive with the deprecation warning.

Let me see what else we can do here, our SDK is not pulling in UIKit anymore, so I'm not sure why UIWebView is being referenced from the dSYM files

**Comment by dylanjha** [comment]
Okay, @markhomoki so I think I figured out what's going on here.

First of all, this sdk `Mux-Stats-AVPlayer` did explicitly import `UIKit` until `1.3.6`: changed in  https://github.com/muxinc/mux-stats-sdk-avplayer/pull/49

Aside from that, this SDK has a dependency on Mux-Stats-Core [specified here](https://github.com/muxinc/mux-stats-sdk-avplayer/blob/master/Mux-Stats-AVPlayer.podspec#L17)

It turns out that `Mux-Stats-Core` imports `UIKit` - and I looked into that core lib and confirmed that it actually uses some of the UIKit classes so that import cannot easily be removed.

Okay, so having said that - **no where in either of these libraries is UIWebView used** that can be confirmed by running the following and getting no matches:

```
grep -r "UIWebView" ./Pods/Mux-Stats-AVPlayer/Frameworks/iOS/fat/MUXSDKStats.framework/
```

The dSYM files are files generated with a reference to UIWebView because we build against iOS version 8+, which means the iOS version we build against *may contain* UIWebView. The dSYM don't actually include any code that runs in your application.

So, given all that I don't really know what else we can do here. Googling around (like the [Mapbox issues above](https://github.com/mapbox/ios-sdk-examples/issues/373)) shows a lot of other people running into this too with other libraries.

My number 1 priority is to work with you here and make sure we're not doing something in our SDK that is going to block your app from being accepted, but it really feels to me like this deprecation warning from apple is overzealous on matching the `dSYM/` files  -- no where is UIWebView being used so it doesn't seem like that would justify a violation.

**Comment by markhomoki** [comment]
Thank you for that.

So I was doing some debugging until night to see what can cause the issue. And I think it was that I took granted that seeing the grep will be automatically rejected by Apple.

I had an idea of trying mux with another project which did not use it before, and also didn't have any deprecation issues. So I installed it and although grep finds `UIWebView`, Apple accepts it without warning.

I appreciate your time and debugging you made, and I'm sorry that I did not have a detailed `try-and-error`s beforehand which at least could have saved some of your time.

Looks like we can close this ticket as this library doesn't use UIWebView. I need to see and find which other lib causes this issue.
